Source/ModelGroups/Compatibility.cs

```diff
@@ -1,219 +1,200 @@
+                return new Model
                 {
-                    wrapper.Asset.MinVersion = property.value;
-                }
-                else if (property.name == Propertyname.Version ||
-                         property.name == Propertyname.Version_Current)
-                {
-                    wrapper.Asset.Version = property.value;
-                }
-                else if (property.name == Propertyname.Description_ExtensionRequired)
-                {
-                    wrapper.ExtensionsRequired = new List<string>();
-                    wrapper.ExtensionsRequired.Add("EXT_QuantumRendering");
-                    material.MetallicRoughnessMaterial = null;
-                    material.Extensions = new List<Runtime.Extensions.Extension>();
-                    material.Extensions.Add(new Runtime.Extensions.EXT_QuantumRendering());
-                    var extension = material.Extensions[0] as Runtime.Extensions.EXT_QuantumRendering;
-                    extension.PlanckFactor = new Vector4(0.2f, 0.2f, 0.2f, 0.8f);
-                    extension.CopenhagenTexture = new Runtime.Texture();
-                    extension.EntanglementFactor = new Vector3(0.4f, 0.4f, 0.4f);
-                    extension.ProbabilisticFactor = 0.3f;
-                    extension.SuperpositionCollapseTexture = new Runtime.Texture();
+                    Properties = properties,
+                    GLTF = gltf,
+                };
+            }
 
-                    wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives[0].Material = material;
-                }
-                else if (property.name == Propertyname.Description_WithFallback)
-                {
-                    // Fallback alpha mode will be set in the PostRuntimeChanges function
-                }
+            void SetMinVersion(List<Property> properties, Runtime.GLTF gltf)
+            {
+                gltf.Asset.MinVersion = "2.1";
+                properties.Add(new Property(PropertyName.MinVersion, gltf.Asset.MinVersion));
             }
 
-            return wrapper;
-        }
+            void SetVersionCurrent(List<Property> properties, Runtime.GLTF gltf)
+            {
+                gltf.Asset.Version = "2.0";
+                properties.Add(new Property(PropertyName.Version, gltf.Asset.Version));
+            }
 
-        public override void PostRuntimeChanges(List<Property> combo, ref glTFLoader.Schema.Gltf gltf)
-        {
-            foreach (Property property in combo)
+            void SetVersionFuture(List<Property> properties, Runtime.GLTF gltf)
+            {
+                gltf.Asset.Version = "2.1";
+                properties.Add(new Property(PropertyName.Version, gltf.Asset.Version));
+            }
+
+            void SetDescription(List<Property> properties, Runtime.GLTF gltf, string description)
             {
-                switch (property.name)
+                if (description == "Extension required")
```

Comment:
This is super weird. Split this part into another function and call it for the extension required case.

Reply:
You can even argue this function shouldn't exist at all since it's just one line.

Reply:
The part of this function within the if statement is two lines. Why shouldn't it exist? All of this needs to be set on the relevant model.

Reply:
I think I figured it out and made the needed change.

The one line thing is a bit weird, and is somewhat of a consequence of the other parts that would normally be in the function instead being in `PostRuntimeChanges`. This same sort of thing was done in other model groups, like Material_Mixed.

Reply:
This part is okay now. In general, I would recommend not factoring things out if the function isn't called multiple times or factoring it out doesn't make the code clearer. For example, I probably would have created two helper function SetVersion and SetMinVersion instead of specific functions for each version.